geniza/corpus/management/commands/googledoc_to_annotation.py

```diff
@@ -0,0 +1,389 @@
+        elif len(metadata) == 8:
+            # rev. 2: includes both Location and Notes fields
+            (pgpid, source_id, location, notes) = (td.get_text() for td in metadata[4:])
+
+        # check if doc name mismatched with pgpid in metadata table; use doc name if so
+        pgpid_match = re.search("PGPID (?P<pgpid>\d+)", name)
+        if pgpid_match and pgpid_match.group("pgpid") != pgpid:
+            pgpid = pgpid_match.group("pgpid")
+
+        # get the document
+        try:
+            doc = Document.objects.get(pk=int(pgpid))
+        except Document.DoesNotExist:
+            print(
+                self.style.WARNING("Document not found for PGPID %s; skipping" % pgpid)
+            )
+            self.document_not_found.append(pgpid)
+            return
+
+        # get the source
+        try:
+            source = Source.objects.get(pk=int(source_id))
+        except Source.DoesNotExist:
+            print(
+                self.style.WARNING(
+                    f"Source not found for Source ID {source_id} (on PGPID {pgpid}); skipping"
+                )
+            )
+            self.source_not_found.append(source_id)
+            return
+
+        # get the first canvas only; researchers want all translation content on first img
+        canvas_base_uri = doc.manifest_uri.replace("manifest", "canvas")
+        iiif_canvas = next(
+            iter(doc.iiif_images(filter_side=True)),
+            f"{canvas_base_uri}1/",  # default in case there are no images
+        )
+
+        # get or create a digital translation footnote
+        (footnote, created) = Footnote.objects.get_or_create(
+            object_id=doc.pk,
+            content_type=self.get_content_type(doc),
+            source=source,
+            doc_relation=Footnote.DIGITAL_TRANSLATION,
+        )
+        footnote.location = location
+        footnote.notes = notes or ""
+        footnote.save()
+        # log creation
+        if created:
+            self.log_addition(
+                footnote,
+                "Created new footnote for migrated Google Docs digital translation",
+            )
+        else:
+            # remove all existing annotations associated with this
+            # document and footnote so we can reimport as needed
+            existing_annos = Annotation.objects.filter(
+                footnote=footnote,
+                created__lt=self.script_run_start,
+            )
+            if existing_annos:
+                print(
+                    "Removing %s pre-existing annotation%s for %s on %s "
+                    % (
+                        len(existing_annos),
+                        pluralize(existing_annos),
+                        footnote.source,
+                        doc.pk,
+                    )
+                )
+                # not creating log entries for deletion, but
+                # this should probably only come up in dev runs
+                existing_annos.delete()
```

Comment:
If this is a dev only scenario lets figure out how to make sure it doesn't happen on a non-dev system. Optional flag to clear out previous imports? rely on debug config to allow this case without an error?